Integrations/integration-Intezer.yml

```diff
@@ -0,0 +1,221 @@
+                default:
+                    throw 'Failed to perform request ' + SERVER_API + url + ', request status code: ' + result.StatusCode + '\n';
+            }
+        }
+        if (result.Body === '' && result.StatusCode === 204) {
+            return { statusCode: result.StatusCode };
+        }
+        if (result.Body === '') {
+            throw 'No content received' + '\n';
+        }
+        var obj;
+        try {
+            obj = JSON.parse(result.Body);
+        } catch (ex) {
+            throw 'Error parsing reply - ' + result.Body + ' - ' + ex;
+        }
+        return { body: result.Body, obj: obj, statusCode: result.StatusCode };
+    };
+
+    var getAnalysis = function(analysisId) {
+        var analysisRes = sendRequest('POST', ANALYSIS_RESULT_URL + analysisId);
+        return analysisRes.obj;
+    };
+
+    var analyzeSHA256 = function(hash, codeItemType, maxRetries, delay) {
+        // validate file is indeed sha256
+        var entries = [];
+        if (!SHA_256_REGEXP.test(hash)) {
+            throw 'File hash is not valid sha256.\nIntezer file hash reputation supports only sha256 hash.\n\n';
+        }
+
+        // create file analysis request
+        var res = sendRequest('POST', HASH_URL, {
+            sha256: hash,
+            code_item_type: codeItemType
+        });
+
+        var analysisId;
+        if (res.statusCode === 201) {
+            analysisId = res.obj.analysis_id;
+        } else {
+            throw 'Failed to create sha256 analysis for ' + hash + ', request status code: ' + res.statusCode + '\n';
+        }
+
+        // perform wait loop
+        res = getAnalysis(analysisId);
+        var tries = 0;
+        while (res.status !== SUCCESS_STATUS && tries < maxRetries) {
+            // this means that status is 'in_progress' or 'queued'
+            tries++;
+            wait(delay); // wait {delay} seconds
+            res = getAnalysis(analysisId);
+        }
+
+        // if polling result did not work, return an error message
+        if (res.status !== SUCCESS_STATUS) {
+            throw 'Failed to pool sha256 analysis for ' + hash + ', Try to change maxRetries or delay arguments\n';
+        } else {
+            // Create new indicator of sha256 according to verdict
+            var verdict = res.result.verdict;
+            var dbotScore = 0;
+            var ec = {};
+            ec[outputPaths.file] = [];
+            if (verdict === 'malicious') {
+                dbotScore = 3;
+                var malFile = {};
+                addMalicious(malFile, outputPaths.file, {
+                    SHA256: hash,
+                    Malicious: {Vendor: 'Intezer', Metadata: res.result}
+                });
+                ec[outputPaths.file].push(malFile[outputPaths.file]);
+            } else if (verdict === 'suspicious') {
+                dbotScore = 2;
+            } else if (verdict === 'trusted' || verdict === 'neutral') {
```

Comment:
is neutral really good from the service point of view? or is it unknown?

Reply:
Yes, worked on that with someone from their team.